packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,225 @@
+import { Tag } from '../cloudformation/tag';
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * Properties Tags is a dictionary of tags as strings
+ */
+export interface Tags {
+    [key: string]: string
+}
+
+/**
+ * A an object of tags with value and properties
+ *
+ * This is used internally but not exported
+ */
+interface FullTags {
+    [key: string]: {value: string, props?: TagProps};
+}
+
+/**
+ * Properties for a tag
+ */
+export interface TagProps {
+    /**
+     * If true all child taggable `Constructs` will receive this tag
+     *
+     * @default true
+     */
+    propagate?: boolean;
+
+    /**
+     * If set propagated tags from parents will overwrite tags in children
+     *
+     * @default false
+     */
+    propagateOverride?: boolean;
+
+    /**
+     * If set this tag will overwrite existing tags
```

Comment:
Also for propagated tags?

Reply:
If you set this `overwrite: true` and `propagateOverride: true`, the tag will be written locally but if a propagating tag is set it will take precedence.

If you set this `overwrite: true` and `propagateOverride: false` (default), the tag will be written locally and will take precedence over the same tag propagated from a parent. I had the classic situation where a single binary value was not enough to address the three use cases:
1. overwrite the local tag
1. do not overwrite the local tag
1. if set now or in the future take the parent tag
1. never take the parent tag

If you combine these possibilities some are illogical but the table is larger than `0` & `1`.

Reply:
Okay, I may have misunderstood the purpose of `propagateOverride`. Here was my assumption:

> `propagateOverride` will control whether we overwrite existing tags while propagating this tag down.

Upon reading your comment and looking at the test, does it actually mean this?

> `propagateOverride` will control whether this tag can be overwritten by tags that are being propagated from parents.

If it's the latter, I would suggest renaming, because this name breaks my brain. May I suggest calling it `sticky` and reversing the meaning (i.e., `sticky=true <=> propagateOverride=false` and vice versa).